src/core/border_router/routing_manager.cpp

```diff
@@ -514,21 +542,34 @@ const Ip6::Prefix *RoutingManager::EvaluateOnLinkPrefix(void)
         {
             otLogInfoBr("EvaluateOnLinkPrefix: There is already smaller on-link prefix %s on interface %u",
                         smallestOnLinkPrefix->ToString().AsCString(), mInfraIfIndex);
-
-            // TODO: we remove the advertised on-link prefix by setting zero valid lifetime for PIO,
-            // But SLAAC addresses configured by PIO prefix will not be removed immediately (
-            // https://tools.ietf.org/html/rfc4862#section-5.5.3). This leads to a situation that
-            // a WiFi device keeps using the old SLAAC address but a Thread device cannot reach to
-            // it. One solution is to delay removing external route until the SLAAC addresses are
-            // actually expired/deprecated.
-            RemoveExternalRoute(*mAdvertisedOnLinkPrefix);
+            DeprecateOnLinkPrefix();
```

Comment:
Just a question to make sure I undertsand the code:
- Here insetad of removing `mAdvertisedOnLinkPrefix ` as an external route immediately, we call `DeprecateOnLinkPrefix()` which starts a timer and from timer callback later we remove `mLocalOnLinkPrefix`.
- Is it then correct that `mAdvertisedOnLinkPrefix` can only be `&mLocalOnLinkPrefix` or `nullptr`.
   - Reading the code, I think this is indeed the case but want to confirm that there is no use-case/situation where it can be some other prefix.
   - If that's the case, can we consider (in a future PR) to change this to be a boolean flag (something like `mIsAdvertisingLocalPrefix` instead of being a pointer)? Does this make sense?

Reply:
@abtink You are right about above points!
I changed to use `mIsAdvertisingLocalOnLinkPrefix` in the third commit https://github.com/openthread/openthread/pull/6878/commits/383f12f4fabbc1f8f08a3d8750f8d562967ca995, PTAL.

Reply:
LGTM. Thanks for taking care of it (also thank for keeping it a separate commit, easy to check and review).